core/src/main/java/org/mule/routing/AbstractSplitter.java

```diff
@@ -33,6 +32,6 @@
         return new CollectionMessageSequence(splitMessage(event));
     }
 
-    protected abstract List<MuleMessage> splitMessage(MuleEvent event) throws MuleException;
+    protected abstract List<MuleEvent> splitMessage(MuleEvent event) throws MuleException;
```

Comment:
That is not clear from the API, as now it takes an event and return a list of events, in the past it took a Message and returned a list of messages. I think the name must change